src/Service/Service.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace TheAentMachine\Service;
+
+use Opis\JsonSchema\ValidationError;
+use Opis\JsonSchema\Validator;
+use TheAentMachine\Service\Exception\ServiceException;
+
+class Service implements \JsonSerializable
+{
+    /** @var string */
+    protected $serviceName;
+    /** @var string */
+    protected $image;
+    /** @var int[]|string[] */
+    protected $internalPorts;
+    /** @var string[] */
+    protected $dependsOn;
+    /** @var mixed[] */
+    protected $ports;
+    /** @var mixed[] */
+    protected $labels;
+    /** @var mixed[] */
+    protected $environment;
+    /** @var mixed[] */
+    protected $volumes;
+    /** @var \stdClass */
+    protected $validatorSchema;
+
+    /**
+     * Service constructor.
+     */
+    public function __construct()
+    {
+        $this->serviceName = '';
+        $this->image = '';
+        $this->internalPorts = array();
+        $this->dependsOn = array();
+        $this->ports = array();
+        $this->labels = array();
+        $this->environment = array();
+        $this->volumes = array();
+        $this->validatorSchema = json_decode(file_get_contents(__DIR__ . '/ServiceJsonSchema.json'), false);
+    }
+
+    /**
+     * @param mixed[] $payload
+     * @return Service
+     * @throws ServiceException
+     */
+    public static function parsePayload(array $payload): Service
+    {
+        $service = new Service();
+        $service->checkValidity($payload);
+        $service->serviceName = $payload['serviceName'] ?? '';
+        $s = $payload['service'] ?? array();
+        if (!empty($s)) {
+            $service->image = $s['image'] ?? '';
+            $service->internalPorts = $s['internalPorts'] ?? array();
+            $service->dependsOn = $s['dependsOn'] ?? array();
+            $service->ports = $s['ports'] ?? array();
+            $service->labels = $s['labels'] ?? array();
+            $service->environment = $s['environment'] ?? array();
+            $service->volumes = $s['volumes'] ?? array();
+        }
+        return $service;
+    }
+
+    /**
+     * @param \stdClass|array|string $data
+     * @param bool $throwsException
+     * @return bool
+     * @throws ServiceException
+     */
+    public function checkValidity($data, bool $throwsException = true): bool
```

Comment:
You can make this property private and remove the $throwsException flag (always throw an exception)